### PR TITLE
[FIX] REST `users.setAvatar` endpoint wasn't allowing update the avatar of other users even with correct permissions

### LINKED ISSUE #11334: [BUG] Unable to set avatar of another profile even with the necessary permission
### Description:
I'm running a small script that updates all RC users avatars from a database. When I fetch the endpoint I get a success message, but the avatar doesn't get updated.

It looks like the server doesn't enter [here](https://github.com/RocketChat/Rocket.Chat/blob/2e38466800edf158d0117bf9cbdc8d1c9568ebf9/packages/rocketchat-lib/server/functions/setUserAvatar.js#L32) or if it does it is not accepting raw Buffers. I have tried changing the avatar with a hosted image (body containing userId and avatarUrl) as in [this example ](https://rocket.chat/docs/developer-guides/rest-api/users/setavatar/#example-call-1)and it does work. The problem is with an image that has been uploaded. I followed the example in the official [RocketChat documentation](https://rocket.chat/docs/developer-guides/rest-api/users/setavatar/#example-call). Nevertheless, looking at the testing folders I found that this option is not tested. The only function tested in Set Avatar is with a [photo from a URL](https://github.com/RocketChat/Rocket.Chat/blob/2e38466800edf158d0117bf9cbdc8d1c9568ebf9/tests/end-to-end/api/01-users.js#L106).

#### Code Used:
```javascript
  const form = new FormData()
  examplePhoto = fs.readFileSync('./example.jpg') //Local image
  form.append('userId', userId) // ID of the user
  form.append('image', examplePhoto) //JPEG file in buffer

  let responseStream = await fetch(`${rocketChatUrl}/api/v1/users.setAvatar`, {
    method: 'POST',
    headers,
    body: form
  })
console.log(responseStream.json()) //Output: {Success: true}
```

### Expected behavior:
When the 'user.setAvatar' endpoint is reached, and a local file is uploaded, the server must send a 'success: true' and the user's avatar will be updated.

### Actual behavior:
When the user.setAvatar endpoint is reached as a POST method with all the parameters required, the server responds with an success 'true' response but the uploaded image file is not actually changed on the user it is being applied to.

### Server Setup Information:

- Version of Rocket.Chat Server: 2.10.2
- Official Rocket Chat image
- Operating System: Ubuntu 17.10
- NodeJS Version: 9.11

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Closes #11334

### CODE REVIEW

File: packages/rocketchat-api/server/v1/users.js [hunk 1]
```diff
@@ -220,24 +220,42 @@ RocketChat.API.v1.addRoute('users.setAvatar', { authRequired: true }, {
...
+				let fields = {};
+				const getUserFromFormData = (fields) => {
+					let userFromFormData;
+					if (fields.userId) {
+						userFromFormData = RocketChat.models.Users.findOneById(fields.userId);
+					} else if (fields.username) {
+						userFromFormData = RocketChat.models.Users.findOneByUsername(fields.username);
+					}
+					return userFromFormData;
+				};
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/users.js]: Suggestion:
```javascript
const getUserFromFormData = (fields) => {
	if (fields.userId) {
		return RocketChat.models.Users.findOneById(fields.userId, {_id: 1});
	}
	if (fields.username) {
		return RocketChat.models.Users.findOneByUsername(fields.username, {_id: 1});
	}
};
```

File: packages/rocketchat-api/server/v1/users.js [hunk 2]
```diff
@@ -220,24 +220,42 @@ RocketChat.API.v1.addRoute('users.setAvatar', { authRequired: true }, {
...
+								}
+							}
 							RocketChat.setUserAvatar(user, Buffer.concat(imageData), mimetype, 'rest');
 							callback();
 						}));
-
 					}));
+					busboy.on('field', (fieldname, val) => {
+						fields = Object.assign(fields, { [fieldname]: val });
+					});
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/users.js]: Why not?:
```javascript
busboy.on('field', (fieldname, val) => {
	fields[fieldname] = val;
});
```

File: packages/rocketchat-api/server/v1/users.js [hunk 3]
```diff
@@ -220,24 +220,42 @@ RocketChat.API.v1.addRoute('users.setAvatar', { authRequired: true }, {
...
-
 						const imageData = [];
 						file.on('data', Meteor.bindEnvironment((data) => {
 							imageData.push(data);
 						}));
 
 						file.on('end', Meteor.bindEnvironment(() => {
+							const sentTheUserByFormData = fields.userId || fields.username;
+							if (sentTheUserByFormData) {
+								user = getUserFromFormData(fields);
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/users.js]: Where are we checking the permissions to change the other user's avatar?

**Comment by GiverOfGifts** [comment]
Any updates on this, why is it on hold?

File: packages/rocketchat-api/server/v1/users.js (hunk 3 above)
Comment (sampaiodiego) [reply packages/rocketchat-api/server/v1/users.js]: at [line 250](https://github.com/RocketChat/Rocket.Chat/pull/11431/files#diff-1d5c7afdaba4728789f33cfa02ec1515R250)

**Pull request merged**